core/src/package_store.rs

```diff
@@ -0,0 +1,559 @@
+            // chosen, only the entries of the list is included (which is none)
+            let (include_entries, include_list) = match import_option {
+                Some(ModuleImportConfig::HideAll) => continue,
+                Some(ModuleImportConfig::ImportAll) => (false, vec![]),
+                Some(ModuleImportConfig::Exclude(vec)) => (false, vec),
+                Some(ModuleImportConfig::Include(vec)) => (true, vec),
+                None => (true, vec![]),
+            };
+            if let Err(e) = Self::insert_transforms(
+                &mut self.transforms,
+                pkg,
+                include_entries,
+                include_list.as_slice(),
+            ) {
+                errors.push(e);
+            }
+        }
+
+        mem::take(&mut config.0)
+            .into_keys()
+            .map(|id| CoreError::UnusedConfig(id.name))
+            .for_each(|e| errors.push(e));
+
+        if errors.is_empty() {
+            Ok(())
+        } else {
+            Err(errors)
+        }
+    }
+
+    // This function was introduced to avoid repeated code. It takes a map and a package, and adds
+    // all transforms in that package which either exists in the list (include_entries=true) or
+    // doesn't exist in the list (include_entries=false) into the map
+    fn insert_transforms(
+        map: &mut HashMap<String, TransformVariant>,
+        pkg: &Package,
+        include_entries: bool,
+        include_list: &[String],
+    ) -> Result<(), CoreError> {
+        for transform @ Transform {
+            from,
+            to,
+            description: _,
+            arguments: _,
+        } in &pkg.info.transforms
+        {
+            if include_entries == include_list.contains(from) {
+                for output_format in to {
+                    if map
+                        .get(from)
+                        .and_then(|t| t.find_transform_to(output_format))
+                        .is_some()
+                    {
+                        return Err(CoreError::OccupiedTransform(
+                            from.clone(),
+                            output_format.to_string(),
+                            pkg.info.name.clone(),
+                        ));
+                    }
+                    let mut target = map
+                        .remove(from)
+                        .unwrap_or_else(|| TransformVariant::External(HashMap::new()));
+                    target.insert_into_external(
+                        output_format.clone(),
+                        (transform.clone(), pkg.clone()),
+                    );
+                    // Add the modified entry back to the map
+                    map.insert(from.clone(), target);
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    pub fn is_missing_packages(&self) -> bool {
+        !self.awaited_packages.is_empty()
+    }
+
+    pub(crate) fn resolve_request(&mut self, request: PackageID, response: Vec<u8>) {
```

Comment:
Maybe we want to use the same terminology as before 
```suggestion
    pub(crate) fn resolve_task(&mut self, id: PackageID, response: Vec<u8>) {
```